File: leetcode/arrays/26-remove-duplicates-from-sorted-array/optimal_solution.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # index starts with 1, cz the first element always unique. (sorted array)
        index = 1
        
        for i in range(len(nums)-1):
            if nums[i] != nums[i+1]:
                nums[index] = nums[i+1]
                index = index + 1
        return index

# Runtime: 80 ms, faster than 91.88% of Python3 online submissions for Remove Duplicates from Sorted Array.
# Memory Usage: 15.6 MB, less than 79.18% of Python3 online submissions for Remove Duplicates from Sorted Array.

File: leetcode/arrays/27-remove-element/optimal-solution.java
class Solution {
    public int removeElement(int[] nums, int val) {
        // OPTIMAL SOLUTION
        
        int index = 0;
        
        for(int i = 0; i < nums.length; i++) {
            if(nums[i] != val)
                nums[index++] = nums[i];
        }
        
        return index;
    }
}

// Runtime: 0 ms, faster than 100.00% of Java online submissions for Remove Element.
// Memory Usage: 37.5 MB, less than 73.85% of Java online submissions for Remove Element.